### DISCUSSION TITLE
Displaying dynamic table on form that updates when multi select options change

### DISCUSSION THREAD

**Discussion by glenelkinsdev** [filamentphp/filament#11813] [Help]
### Package

Form builder

### Package Version

3

### How can we help you?

I'm using the admin panel and i have a resource with a form. On the form is a multi select which is a relationship, so in this case my create form is for Product model and the multi select is a relationship with Ingredient model on a many-to-many

I need a table at the top of the form that when these options change the info for the selected products is displayed in the table, so it takes the selected ingredient ids and pulls the data from db for them.

The only way i've found i can do this so far is as follows:

1) Set the multi-select to live (using ->live())
2) Create a custom form component which actually just has a table displaying data from an array in the component
3) Use the component in the form as a field and pass the form object to the component and get the state of the select fields for the ids

Is there a better way of doing this? 

I did think of using a repeater for the ingredients, the problem with that is that is that i wouldn't want the ingredient created every time a new product is added, i'd want to have it so the repeater allows them to select an existing product OR add a new one but i'm not sure how to get the repeater doing that (so it understands if an ingredient is selected in the dropdown just link the product to it, otherwise use the fields to create a new ingredient)

**Comment by micraux** [comment]
Maybe a `RelationManager`?

You can Attach/Detach ingredients and even Create a new ingredient in case it doesn't exist...
